CT-1647 add docker container for uploads

misaka opened the pull request:
Changes in this PR include:

- Support an additional Docker container called "uploads" where PDF generation and anti-virus will go.
- Rename "reports generator" to just "reports" (this was already half-done in the deploy repo).
- Add `docker-compose.yml` so that we can run the Docker images locally.
- Remove need for base image generation.

The changes here will require a PR in the `correspondence-staff-deploy` repo and changes to Jenkins. See the PR https://github.com/ministryofjustice/correspondence-staff-deploy/pull/48 for more details.

misaka commented on app/controllers/cases_controller.rb:
```diff
@@ -134,8 +140,13 @@ def set_permitted_events
+      :date_responded_yyyy,
+      :outcome_name,
+      :refusal_reason_name,
+      exemption_ids: params[:case][:exemption_ids].nil? ? nil : params[:case][:exemption_ids].keys
```
Looks like a prime candidate for Ruby's new "safe-navigation" operator:

```ruby
exemption_ids: params[:case][:exemption_ids]&.keys
```

aliuk2012 commented:
From a frontend point of view, we will need to refactor the markup to not use br tags and to use the appropriate tags and move all page copy to locales file. I'll create a new ticket for us to redo those

stephenrichards commented:
That's cool, and what I expected.  Just wanted to get something out there
as soon as possible.

On 10 March 2017 at 20:51, Alistair Laing <notifications@github.com> wrote:

> From a frontend point of view, we will need to refactor the markup to not
> use br tags and to use the appropriate tags and move all page copy to
> locales file. I'll create a new ticket for us to redo those
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ministryofjustice/correspondence_tool_staff/pull/119#issuecomment-285781157>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHDod4R1k78NwEgoz82e4exriCvkHn9ks5rkbfLgaJpZM4MZiI1>
> .
>



-- 
Stephen Richards

misaka committed:
CT-1621 add redis to docker-compose, development Dockerfile

misaka committed:
CT-1621 drop app-base image, create & migrate when running web

misaka committed:
CT-976 add mailer, reports and uploads images to docker-compose

Also use DATABASE_URL in database.yml for production env

misaka committed:
CT-976 setup uploads docker image and scripts

A couple house-keeping items packaged into this commit:

- rename "report-generator" image to just "reports"
- reconfigure sidekiq concurrency, generally reducing concurrent threads since
  we're adding sidekiq processes

misaka committed:
CT-1621 use DATABASE_URL in development env when present

Allows us to conditionally connect to localhost when running in development mode
locally, or if we're running in development using docker-compose connect do the
db service that we spin up.

misaka committed:
CT-1621 don't install PSQL onto uploads-base, no real need

We don't install psql onto the other sidekiq containers for this project, so for
consistency. It also speeds up the build, but since this base image is built
separately from when we build the project (typically scheduled in Jenkins in off
hours) this isn't really significant.

misaka committed:
CT-1621 use DATABASE_URL in development env for all the services

Turns out I'm not sure how to set DATABASE_URL in the deploy repo (because of
the password), so it seems like a better pattern to start all services in
development mode (in docker-compose) and to only use DATABASE_URL in the
development section of database.yml

misaka committed:
CT-1621 merge all Dockerfiles into one and use official ruby images

By switching to the official ruby images as a base for our application image
we've gotten around issues that were invalidating layer caches, which has
allowed us to construct a single Dockerfile that can be responsible for building
all of our images.

misaka committed:
CT-1647 change CMD to ENTRYPOINT in Dockerfile

This was changed when I didn't understand the difference between entrypoint and
cmd.

misaka committed:
CT-1647 update README and move development_mode in Dockerfile

misaka committed:
CT-1647 add freshclam to docker-compose.yml

misaka committed:
CT-1647 fix clamav-freshclam typo

misaka committed:
CT-1647 fix entrypoint overrides in docker-compose.yml

misaka committed:
CT-1647 seed dev teams and users when running in dev mode

misaka committed:
CT-1647 fix development mode when running bundle install

misaka committed:
CT-1647 make db data persistent in docker container

misaka committed:
CT-1647 remove 'touch /etc/inittab' from Dockerfile

Don't know why we would need this and it seems to be working fine with it
commented out.

misaka committed:
CT-1647 disable rubocop method length check in seeders
